[BUG] main ブランチへのマージ時に実行される Azure への deploy が失敗することがある

### 概要

表題通り。

- 失敗したときの事例
https://github.com/digitaldemocracy2030/kouchou-ai/actions/runs/20540103407/attempts/1

- リトライで直った
https://github.com/digitaldemocracy2030/kouchou-ai/actions/runs/20540103407

### 再現手順

1. main ブランチに更新を入れる
2. GitHub Actions で Azure への deploy 処理が動く
3. たまに失敗する

### 期待する動作

毎回 deploy が成功する
<!-- 本来どう動作すべきかを記入してください -->

### スクリーンショット・ログ

<!-- 必要に応じてスクリーンショットやエラーログなどを添付してください -->

### その他

AIによると、原因はこんな感じ
```
発生箇所: Azure デプロイの Docker ビルド中、client イメージの Dockerfile:14 の RUN npm ci で npm ERR! code ECONNRESET により失敗。内容は npm レジストリへの接続リセットで、依存関係の不整合ではありません。
文脈: azure-deploy.yml:53-90 で API/client/client-admin/client-static-build の 4 イメージを並列ビルド。複数の npm ci が同時に走るため、GitHub ホストランナー上でレジストリへの接続がまれに途切れる典型的なネットワーク起因の失敗です。
他のステージ (Python 依存、client-admin ビルドなど) は完走しており、恒常的な依存問題ではなく一時的な接続切れが原因と考えられます。

対応案（優先度順）

1. まずはリトライ実行: ワークフローを再実行すれば通る可能性が高いです。
2. npm ci にリトライとタイムアウトを付与 (各 Dockerfile の npm 実行直前で):
例: npm config set fetch-retries 5 && npm config set fetch-retry-mintimeout 20000 && npm config set fetch-retry-maxtimeout 120000 && npm config set registry https://registry.npmjs.org/
あるいは npm ci --fetch-retries=5 --fetch-retry-mintimeout=20000 --fetch-retry-maxtimeout=120000 --network-timeout=600000
対象: Dockerfile:14, Dockerfile:17, Dockerfile:5 と同ファイル内 2 回目の npm ci.
3. 並列度を下げる: ワークフローで Docker ビルドを直列にする、または client 系をまとめて後段にずらすことでレジストリへの同時接続を減らす。
4. npm キャッシュ活用: Docker BuildKit のキャッシュマウントや npm ci --cache /tmp/npm-cache を併用し、外部アクセス回数を削減。
まずは 1) を試し、再発するようなら 2) を反映、それでも改善しなければ 3)/4) を検討するのが現実的です。
```

ここのところ毎回 deploy に失敗している模様

https://github.com/digitaldemocracy2030/kouchou-ai/actions/workflows/azure-deploy.yml

<img width="1098" height="401" alt="Image" src="https://github.com/user-attachments/assets/7138ad91-2892-48e5-9cf2-863b1218f52e" />

失敗し始めた job のログはこんな感じ。ディレクトリ構成が変わったことに起因してそう（まだちゃんと見れてない）
https://github.com/digitaldemocracy2030/kouchou-ai/actions/runs/21274113788/job/61230336520
```
#8 [builder  8/10] COPY packages/report-schema packages/report-schema
#10 CACHED
#8 ERROR: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::wd3cekjid0skg8qo5fedbha0c: "/packages/report-schema": not found


#11 [ 3/10] RUN apt-get update     && apt-get install -y --no-install-recommends         curl         build-essential     && apt-get clean     && rm -rf /var/lib/apt/lists/*
#9 [builder  9/10] COPY apps/public-viewer apps/public-viewer
#11 CACHED
#9 ERROR: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::wd3cekjid0skg8qo5fedbha0c: "/apps/public-viewer": not found


#12 [ 4/10] RUN pip install uv
#10 [builder  6/10] COPY apps/public-viewer/package.json apps/public-viewer/
#10 ERROR: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::wd3cekjid0skg8qo5fedbha0c: "/apps/public-viewer/package.json": not found
#12 CACHED


#11 [builder  4/10] COPY package.json pnpm-lock.yaml pnpm-workspace.yaml .npmrc ./
#13 [ 7/10] RUN uv pip install --no-cache --system /packages/analysis-core
#11 ERROR: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::wd3cekjid0skg8qo5fedbha0c: "/pnpm-workspace.yaml": not found

#13 CACHED
#12 [builder  5/10] COPY packages/report-schema/package.json packages/report-schema/

#12 ERROR: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::wd3cekjid0skg8qo5fedbha0c: "/packages/report-schema/package.json": not found
#14 [10/10] COPY apps/api/ .
#14 ERROR: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::xxoyvgm9z67v8qjsen0k2nvkc: "/apps/api": not found


#15 [ 1/10] FROM docker.io/library/python:3.12-slim@sha256:5e2dbd4bbdd9c0e67412aea9463906f74a22c60f89eb7b5bbb7d45b66a2b68a6
#13 [builder  1/11] FROM docker.io/library/node:22-alpine@sha256:a9cd9bac76cf2396abf14ff0d1c3671a8175fe577ce350e62ab0fc1678050176
#13 resolve docker.io/library/node:22-alpine@sha256:a9cd9bac76cf2396abf14ff0d1c3671a8175fe577ce350e62ab0fc1678050176 done
#15 resolve docker.io/library/python:3.12-slim@sha256:5e2dbd4bbdd9c0e67412aea9463906f74a22c60f89eb7b5bbb7d45b66a2b68a6 0.0s done
------
 > [builder  4/10] COPY package.json pnpm-lock.yaml pnpm-workspace.yaml .npmrc ./:
------
------
#15 DONE 0.0s
 > [builder  5/10] COPY packages/report-schema/package.json packages/report-schema/:
------
------
 > [builder  6/10] COPY apps/public-viewer/package.json apps/public-viewer/:
------
------
------
 > [builder  8/10] COPY packages/report-schema packages/report-schema:
 > [10/10] COPY apps/api/ .:
------
------
------
 > [builder  9/10] COPY apps/public-viewer apps/public-viewer:
------

#3 [internal] load .dockerignore
#3 transferring context: 385B done
#3 DONE 0.0s

#4 [internal] load build context
#4 transferring context: 1.30kB done
#4 DONE 0.0s

#5 [builder  3/11] RUN corepack enable
#5 CACHED

#6 [builder  7/11] RUN pnpm install --frozen-lockfile
#6 CACHED

#7 [builder  2/11] WORKDIR /repo
#7 CACHED

#8 [builder  9/11] COPY apps/admin apps/admin
#8 ERROR: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::unytqryicj2s0iq17rzsoctv1: "/apps/admin": not found

#9 [builder  8/11] COPY packages/report-schema packages/report-schema
#9 ERROR: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::unytqryicj2s0iq17rzsoctv1: "/packages/report-schema": not found

#10 [builder  6/11] COPY apps/admin/package.json apps/admin/
#10 ERROR: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::unytqryicj2s0iq17rzsoctv1: "/apps/admin/package.json": not found

#11 [builder  5/11] COPY packages/report-schema/package.json packages/report-schema/
#11 ERROR: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::unytqryicj2s0iq17rzsoctv1: "/packages/report-schema/package.json": not found

#12 [builder  4/11] COPY package.json pnpm-lock.yaml pnpm-workspace.yaml .npmrc ./
#12 ERROR: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::unytqryicj2s0iq17rzsoctv1: "/pnpm-workspace.yaml": not found

#13 [builder  1/11] FROM docker.io/library/node:22-alpine@sha256:a9cd9bac76cf2396abf14ff0d1c3671a8175fe577ce350e62ab0fc1678050176
#13 resolve docker.io/library/node:22-alpine@sha256:a9cd9bac76cf2396abf14ff0d1c3671a8175fe577ce350e62ab0fc1678050176 done
------
 > [builder  4/11] COPY package.json pnpm-lock.yaml pnpm-workspace.yaml .npmrc ./:
------
------
 > [builder  5/11] COPY packages/report-schema/package.json packages/report-schema/:
------
------
 > [builder  6/11] COPY apps/admin/package.json apps/admin/:
------
------
 > [builder  8/11] COPY packages/report-schema packages/report-schema:
------
------
 > [builder  9/11] COPY apps/admin apps/admin:
------

#4 [internal] load .dockerignore
#4 transferring context: 1.34kB done
#4 DONE 0.0s

#5 [internal] load build context
#5 transferring context: 953.35kB 0.0s done
#5 DONE 0.0s

#6 [builder  1/11] FROM docker.io/library/node:22-alpine@sha256:a9cd9bac76cf2396abf14ff0d1c3671a8175fe577ce350e62ab0fc1678050176
#6 resolve docker.io/library/node:22-alpine@sha256:a9cd9bac76cf2396abf14ff0d1c3671a8175fe577ce350e62ab0fc1678050176 done
#6 sha256:1074353eec0db2c1d81d5af2671e56e00cf5738486f5762609ea33d606f88612 0B / 3.86MB 0.1s
#6 sha256:27dda044491a389c13d0aa53e5cc14375121749b7401b4db1825fd97191528aa 7.34MB / 51.60MB 0.1s
#6 sha256:0a5453a6a95efdb1bc3321878ea844e2c518f39aecd5f7f8a78e1b78d89a637e 1.26MB / 1.26MB 0.1s done
#6 sha256:cd007ec038c44249b9612b09663aa39b77c77bf71eeed265f47bf14154bce59c 0B / 446B 0.1s
#6 sha256:a9cd9bac76cf2396abf14ff0d1c3671a8175fe577ce350e62ab0fc1678050176 6.41kB / 6.41kB done
#6 sha256:bafc917fc7678d44c26d30e164b012e9144cc014ad73e6d97ed6a822b4e74d5e 1.72kB / 1.72kB done
#6 sha256:ff729bb6df6b990a022dca2a6aea455c7c89a85eabd1692684ca21bbbe0a78a9 6.52kB / 6.52kB done
Dockerfile:20
--------------------
  18 |     
  19 |     COPY packages/report-schema packages/report-schema
  20 | >>> COPY apps/public-viewer apps/public-viewer
  21 |     
  22 |     RUN pnpm --filter @kouchou-ai/report-schema build
--------------------
ERROR: failed to build: failed to solve: failed to compute cache key: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::wd3cekjid0skg8qo5fedbha0c: "/apps/public-viewer": not found
Dockerfile:43
--------------------
  41 |     
  42 |     # Copy application source
  43 | >>> COPY apps/api/ .
  44 |     
  45 |     # アプリケーションの実行
--------------------
ERROR: failed to build: failed to solve: failed to compute cache key: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::xxoyvgm9z67v8qjsen0k2nvkc: "/apps/api": not found
Dockerfile:23
--------------------
  21 |     
  22 |     COPY packages/report-schema packages/report-schema
  23 | >>> COPY apps/admin apps/admin
  24 |     
  25 |     ENV NEXT_DISABLE_FETCH_DURING_BUILD=true
--------------------
ERROR: failed to build: failed to solve: failed to compute cache key: failed to calculate checksum of ref 6d647f4b-a064-4489-8a08-7cb8635b8d79::unytqryicj2s0iq17rzsoctv1: "/apps/admin": not found
#6 sha256:1074353eec0db2c1d81d5af2671e56e00cf5738486f5762609ea33d606f88612 3.86MB / 3.86MB 0.1s done
#6 sha256:27dda044491a389c13d0aa53e5cc14375121749b7401b4db1825fd97191528aa 28.31MB / 51.60MB 0.2s
#6 sha256:cd007ec038c44249b9612b09663aa39b77c77bf71eeed265f47bf14154bce59c 446B / 446B 0.1s done
#6 extracting sha256:1074353eec0db2c1d81d5af2671e56e00cf5738486f5762609ea33d606f88612 0.1s
#6 sha256:27dda044491a389c13d0aa53e5cc14375121749b7401b4db1825fd97191528aa 51.60MB / 51.60MB 0.3s done
#6 extracting sha256:1074353eec0db2c1d81d5af2671e56e00cf5738486f5762609ea33d606f88612 0.1s done
#6 extracting sha256:27dda044491a389c13d0aa53e5cc14375121749b7401b4db1825fd97191528aa
#6 extracting sha256:27dda044491a389c13d0aa53e5cc14375121749b7401b4db1825fd97191528aa 1.2s done
#6 extracting sha256:0a5453a6a95efdb1bc3321878ea844e2c518f39aecd5f7f8a78e1b78d89a637e
#6 extracting sha256:0a5453a6a95efdb1bc3321878ea844e2c518f39aecd5f7f8a78e1b78d89a637e 0.0s done
#6 extracting sha256:cd007ec038c44249b9612b09663aa39b77c77bf71eeed265f47bf14154bce59c done
#6 DONE 1.9s

#7 [builder  2/11] WORKDIR /repo
#7 DONE 0.0s

#8 [builder  3/11] RUN corepack enable
#8 DONE 0.2s

#9 [builder  4/13] COPY package.json pnpm-lock.yaml pnpm-workspace.yaml .npmrc ./
#9 DONE 0.0s

#10 [builder  5/13] COPY packages/report-schema/package.json packages/report-schema/
#10 DONE 0.0s

#11 [builder  6/13] COPY apps/static-site-builder/package.json apps/static-site-builder/
#11 DONE 0.0s

#12 [builder  7/13] COPY apps/public-viewer/package.json apps/public-viewer/
#12 DONE 0.0s

#13 [builder  8/13] RUN pnpm install --frozen-lockfile
#13 0.170 ! Corepack is about to download https://registry.npmjs.org/pnpm/-/pnpm-9.15.4.tgz
#13 1.277 Scope: all 4 workspace projects
#13 1.355 Lockfile is up to date, resolution step is skipped
#13 1.434 Progress: resolved 1, reused 0, downloaded 0, added 0
#13 1.601 Packages: +958
#13 1.601 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
#13 1.824 
#13 1.824    ╭───────────────────────────────────────────────────────────────────╮
#13 1.824    │                                                                   │
#13 1.824    │                Update available! 9.15.4 → 10.28.1.                │
#13 1.824    │   Changelog: https://github.com/pnpm/pnpm/releases/tag/v10.28.1   │
#13 1.824    │         Run "corepack install -g pnpm@10.28.1" to update.         │
#13 1.824    │                                                                   │
#13 1.824    ╰───────────────────────────────────────────────────────────────────╯
#13 1.824 
#13 2.435 Progress: resolved 958, reused 0, downloaded 74, added 72
Error: Process completed with exit code 1.
```

--- a/.github/workflows/azure-deploy.yml
+++ b/.github/workflows/azure-deploy.yml
@@ -5,6 +5,10 @@ on:
     branches: [main]
   workflow_dispatch:
 
+concurrency:
+  group: azure-deploy-${{ github.ref }}
+  cancel-in-progress: false
+
 env:
   AZURE_ACR_NAME: ${{ vars.AZURE_ACR_NAME }}
   AZURE_RESOURCE_GROUP: ${{ vars.AZURE_RESOURCE_GROUP }}

